modules/azure/loadbalancer.go

```diff
@@ -0,0 +1,201 @@
+func GetLoadBalancerConfigNames(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) []string {
+	configName, err := GetLoadBalancerConfigNamesE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return configName
+}
+
+// GetLoadBalancerConfigNamesE gets a list of the Frontend Configuration Names for the Load Balancer.
+func GetLoadBalancerConfigNamesE(loadBalancerName string, resourceGroupName string, subscriptionID string) ([]string, error) {
+	var configNames []string
+
+	lb, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return configNames, err
+	}
+
+	lbProps := lb.LoadBalancerPropertiesFormat
+	feConfigs := *lbProps.FrontendIPConfigurations
+	if len(feConfigs) == 0 {
+		// No Frontend Configurations present
+		return configNames, NewNotFoundError("Frontend Config", "Any", loadBalancerName)
+	}
+
+	// Get the names of the Frontend IP Configurations present
+	for _, config := range feConfigs {
+		configNames = append(configNames, *config.Name)
+	}
+
+	return configNames, nil
+}
+
+// GetLoadBalancerFrontendConfig gets the IP address and specifies public or private for the specified Load Balancer.
+// This function would fail the test if there is an error.
+func GetLoadBalancerFrontendConfig(t testing.TestingT, feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType) {
+	ipAddress, ipType, err := GetLoadBalancerFrontendConfigE(feConfigName, loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return ipAddress, ipType
+}
+
+// GetLoadBalancerFrontendConfigE gets the IP address and specifies public or private for the specified Load Balancer.
+func GetLoadBalancerFrontendConfigE(feConfigName string, loadBalancer01Name string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType, err1 error) {
+	// Get the specified Load Balancer Frontend Config
+	feConfig, err := GetLoadBalancerFrontendIPConfigurationE(feConfigName, loadBalancer01Name, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", -1, err
+	}
+
+	// Get the Properties of the Frontend Configuration
+	feProps := *feConfig.FrontendIPConfigurationPropertiesFormat
+
+	// Check Public Type
+	if feProps.PublicIPAddress != nil {
+		// Get PublicIPAddressResource name for Load Balancer Frontend Configuration
+		pipName := GetNameFromResourceID(*feProps.PublicIPAddress.ID)
+
+		ipValue, err := GetIPOfPublicIPAddressByNameE(pipName, resourceGroupName, subscriptionID)
+		if err != nil {
+			return "", -1, err
+		}
+
+		// Public IP
+		return ipValue, IPType(PublicIP), nil
+	}
+
+	// Private IP
+	return *feProps.PrivateIPAddress, IPType(PrivateIP), nil
+
+}
+
+// GetLoadBalancerFrontendIPConfigurationE gets a Load Balancer Configuration in the specified Azure Resource Group
```

Comment:
func name should be GetLoadBalancerFrontendIPConfiguration

Reply:
Also, this needs an additional wrapper method exposed that has no E in name GetLoadBalancerFrontendIPConfiguration.

Reply:
The tests do not call the LoadBalancerFrontEndConfig directly so left the resource call out. The module uses the E version internally. Please advise and can add it.

Reply:
Added